dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/Support/TestStore.cs

```diff
@@ -26,7 +27,24 @@ public abstract class TestStore
     protected virtual Task StopAsync()
         => Task.CompletedTask;
 
-    public abstract VectorStore DefaultVectorStore { get; }
+    public VectorStore DefaultVectorStore
+    {
+        get => this._defaultVectorStore ?? throw new InvalidOperationException("Not initialized");
+        set => this._defaultVectorStore = value;
+    }
+
+#pragma warning disable CA1816, CA1063 // Dispose methods should call SuppressFinalize: no finalizer
+    public void Dispose() => this.Dispose(true);
+#pragma warning restore CA1063, CA1063
+
+    protected virtual void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            this._lock.Dispose();
+            this._defaultVectorStore?.Dispose();
```

Comment:
As in the above comments, am not too sure about this - the default vector store gets instantiated in implementing classes and in StartAsync(). At least in theory, Start/Stop could be called multiple times, so I'd expect disposal of the default vector store to happen in StopAsync in the implementing classes.